diff -urN renderproto-0.11.1/config.sub renderproto-0.11.1-patched/config.sub
--- renderproto-0.11.1/config.sub	2010-08-10 16:11:53.000000000 +0200
+++ renderproto-0.11.1-patched/config.sub	2019-08-26 14:47:29.639728121 +0200
@@ -1302,7 +1302,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -qword*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
