diff -urN libXfont2-2.0.3/config.sub libXfont2-2.0.3-patched/config.sub
--- libXfont2-2.0.3/config.sub	2017-11-28 15:29:02.000000000 +0100
+++ libXfont2-2.0.3-patched/config.sub	2019-08-26 15:55:10.618709924 +0200
@@ -1399,7 +1399,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -tirtos*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -tirtos* | -qword*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -urN libXfont2-2.0.3/src/FreeType/ftenc.c libXfont2-2.0.3-patched/src/FreeType/ftenc.c
--- libXfont2-2.0.3/src/FreeType/ftenc.c	2017-11-28 15:28:56.000000000 +0100
+++ libXfont2-2.0.3-patched/src/FreeType/ftenc.c	2019-08-26 16:24:37.461584720 +0200
@@ -41,6 +41,8 @@
 #include FT_XFREE86_H
 #include "ft.h"
 
+#include <strings.h>
+
 static int find_cmap(int, int, int, FT_Face, FT_CharMap *);
 
 static int
diff -urN libXfont2-2.0.3/src/FreeType/xttcap.c libXfont2-2.0.3-patched/src/FreeType/xttcap.c
--- libXfont2-2.0.3/src/FreeType/xttcap.c	2017-11-28 15:28:56.000000000 +0100
+++ libXfont2-2.0.3-patched/src/FreeType/xttcap.c	2019-08-26 16:26:29.479583171 +0200
@@ -47,6 +47,8 @@
 #include <ctype.h>
 #include <math.h>
 
+#include <strings.h>
+
 #ifndef True
 #define True (-1)
 #endif /* True */
diff -urN libXfont2-2.0.3/src/fc/fsio.c libXfont2-2.0.3-patched/src/fc/fsio.c
--- libXfont2-2.0.3/src/fc/fsio.c	2017-11-28 15:28:56.000000000 +0100
+++ libXfont2-2.0.3-patched/src/fc/fsio.c	2019-08-26 16:46:44.472566370 +0200
@@ -64,6 +64,14 @@
 #endif
 
 
+#ifdef FD_SET
+#undef FD_SET
+#endif
+#ifdef FD_ISSET
+#undef FD_ISSET
+#endif
+#include <sys/select.h>
+
 
 static int  padlength[4] = {0, 3, 2, 1};
 
diff -urN libXfont2-2.0.3/src/fc/fslibos.h libXfont2-2.0.3-patched/src/fc/fslibos.h
--- libXfont2-2.0.3/src/fc/fslibos.h	2017-11-28 15:28:56.000000000 +0100
+++ libXfont2-2.0.3-patched/src/fc/fslibos.h	2019-08-26 16:32:59.111577783 +0200
@@ -71,7 +71,7 @@
 #      ifdef NOFILE
 #       define OPEN_MAX NOFILE
 #      else
-#       define OPEN_MAX NOFILES_MAX
+#       define OPEN_MAX 256
 #      endif
 #     endif
 #    endif
diff -urN libXfont2-2.0.3/src/fontfile/catalogue.c libXfont2-2.0.3-patched/src/fontfile/catalogue.c
--- libXfont2-2.0.3/src/fontfile/catalogue.c	2017-11-28 15:28:56.000000000 +0100
+++ libXfont2-2.0.3-patched/src/fontfile/catalogue.c	2019-08-26 16:22:19.676586625 +0200
@@ -36,6 +36,8 @@
 #include <dirent.h>
 #include <unistd.h>
 
+#include <strings.h>
+
 static const char CataloguePrefix[] = "catalogue:";
 
 static int CatalogueFreeFPE (FontPathElementPtr fpe);
diff -urN libXfont2-2.0.3/src/fontfile/renderers.c libXfont2-2.0.3-patched/src/fontfile/renderers.c
--- libXfont2-2.0.3/src/fontfile/renderers.c	2017-11-28 15:28:56.000000000 +0100
+++ libXfont2-2.0.3-patched/src/fontfile/renderers.c	2019-08-26 16:07:50.499699417 +0200
@@ -34,6 +34,8 @@
 #include "libxfontint.h"
 #include <X11/fonts/fntfilst.h>
 
+#include <strings.h>
+
 static FontRenderersRec	renderers;
 
 /*
