diff -urN libfontenc-1.1.4/config.sub libfontenc-1.1.4-patched/config.sub
--- libfontenc-1.1.4/config.sub	2019-02-20 02:32:58.000000000 +0100
+++ libfontenc-1.1.4-patched/config.sub	2019-08-26 16:01:01.562705072 +0200
@@ -1390,7 +1390,7 @@
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
 	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme* \
-	      | -midnightbsd*)
+	      | -midnightbsd* | -qword*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
