diff -urN xtrans-1.2.7/config.sub xtrans-1.2.7-patched/config.sub
--- xtrans-1.2.7/config.sub	2012-03-23 04:04:40.000000000 +0100
+++ xtrans-1.2.7-patched/config.sub	2019-08-26 06:24:45.322129648 +0200
@@ -1325,7 +1325,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -qword*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
