diff -urN sed-4.7/build-aux/config.sub sed-4.7-patched/build-aux/config.sub
--- sed-4.7/build-aux/config.sub	2018-12-16 19:35:18.000000000 +0100
+++ sed-4.7-patched/build-aux/config.sub	2019-08-11 16:18:44.066850352 +0200
@@ -1361,7 +1361,7 @@
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
 	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
-	     | midnightbsd* | amdhsa* | unleashed*)
+	     | midnightbsd* | amdhsa* | unleashed* | qword*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	qnx*)
