diff -urN xproto-7.0.31/config.sub xproto-7.0.31-patched/config.sub
--- xproto-7.0.31/config.sub	2006-07-01 09:10:25.000000000 +0200
+++ xproto-7.0.31-patched/config.sub	2019-08-26 09:33:42.969290884 +0200
@@ -1217,7 +1217,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos*)
+	      | -skyos* | -haiku* | -rdos* | -qword*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
