diff -urN a/config.sub libpng-1.6.35/config.sub
--- a/config.sub	2019-08-15 21:50:39.177660411 +0200
+++ libpng-1.6.35/config.sub	2018-07-16 05:58:00.000000000 +0200
@@ -1390,7 +1390,7 @@
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
 	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme* \
-	      | -midnightbsd*)
+	      | -midnightbsd* | -qword*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
