diff -urN libX11-1.6.8/config.sub libX11-1.6.8-patched/config.sub
--- libX11-1.6.8/config.sub	2019-06-17 16:47:47.000000000 +0200
+++ libX11-1.6.8-patched/config.sub	2019-08-26 10:34:35.145268369 +0200
@@ -1427,7 +1427,7 @@
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
-	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox*)
+	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -qword*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -urN libX11-1.6.8/src/GetDflt.c libX11-1.6.8-patched/src/GetDflt.c
--- libX11-1.6.8/src/GetDflt.c	2019-06-17 16:47:35.000000000 +0200
+++ libX11-1.6.8-patched/src/GetDflt.c	2019-08-26 11:40:15.658941815 +0200
@@ -65,6 +65,7 @@
 #include <stdio.h>
 #include <ctype.h>
 
+#include <assert.h>
 
 /*ARGSUSED*/
 static char *
@@ -104,6 +105,7 @@
 	(void) strncpy(dest, ptr, len-1);
 	dest[len-1] = '\0';
     } else {
+  assert(!"$HOME variable not set");/*
 	if ((ptr = getenv("USER")))
 	    pw = _XGetpwnam(ptr,pwparams);
 	else
@@ -112,7 +114,7 @@
 	    (void) strncpy(dest, pw->pw_dir, len-1);
 	    dest[len-1] = '\0';
 	} else
-	    *dest = '\0';
+	    *dest = '\0';*/
     }
 #endif
     return dest;
