From 145979874a7ee3b9e4785969fb1f76992488526c Mon Sep 17 00:00:00 2001
From: mintsuki <mintsuki@protonmail.com>
Date: Mon, 20 Jul 2020 10:53:50 +0200
Subject: [PATCH] qword specific changes

---
 config.sub | 2 +-
 configure  | 4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/config.sub b/config.sub
index 00f68b8..52eb178 100755
--- a/config.sub
+++ b/config.sub
@@ -1416,7 +1416,7 @@ case $os in
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
-	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox*)
+	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -qword*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff --git a/configure b/configure
index adead92..d1209ba 100755
--- a/configure
+++ b/configure
@@ -6335,6 +6335,10 @@ echo "${ECHO_T}$cf_cv_ldflags_search_paths_first" >&6
 
 		MK_SHARED_LIB='${CC} ${LDFLAGS} ${CFLAGS} -shared -Wl,-soname,'$cf_cv_shared_soname',-stats,-lc -o $@'
 		;;
+	(qword*)
+		CC_SHARED_OPTS='-fPIC'
+                MK_SHARED_LIB='${CC} -shared -o $@'
+		;;
 	(mingw*)
 		cf_cv_shlib_version=mingw
 		cf_cv_shlib_version_infix=mingw
-- 
2.27.0

