diff -urN automake-1.15.1/lib/config.sub automake-1.15.1-patched/lib/config.sub
--- automake-1.15.1/lib/config.sub	2017-06-18 04:37:14.000000000 +0200
+++ automake-1.15.1-patched/lib/config.sub	2020-01-03 18:24:02.433177798 +0100
@@ -1417,7 +1417,8 @@
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
-	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox*)
+	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* \
+	      | -qword*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
